ARCHITECTURE.md

```diff
@@ -170,3 +188,49 @@ one point.
 
 The prefix `seal` here is for historic reasons. There is some analogy to sealing a
 contract. And we found seals to be a cute animal as well ‒ like squids!
+
+## `Environment` Trait
```

Comment:
I think it's fine to generally mention this here,  but more with an emphasis on the interface instead of details. E.g "this is the way ink! determines concretes types since Substrate types are generic blah blah...".